File: solutions/javascript/scrabble-score/scrabble-score.js
//@ts-check

/**
 * Compute a word's Scrabble score by summing the values of its letters.
 *
 * @param {string} input
 * @returns {number}
 */
export const score = (input) => {
  let sum = input
    .replace(/[^A-Z]/gi, "")
    .replace(/[AEIOULNRST]/gi, "+1")
    .replace(/[DG]/gi, "+2")
    .replace(/[BCMP]/gi, "+3")
    .replace(/[FHVWY]/gi, "+4")
    .replace(/[K]/gi, "+5")
    .replace(/[JX]/gi, "+8")
    .replace(/[QZ]/gi, "+10");

  return eval("0" + sum);
};

File: solutions/python/scrabble-score/scrabble_score.py
"""
A word's Scrabble score is the sum of its letters' values.
"""

from enum import IntEnum


class LetterValue(IntEnum):
    """
    Each letter has a value.
    """

    A = E = I = O = U = L = N = R = S = T = 1
    D = G = 2
    B = C = M = P = 3
    F = H = V = W = Y = 4
    K = 5
    J = X = 8
    Q = Z = 10


def score(word: str) -> int:
    """
    Compute a word's score by summing the values of its letters.
    """
    return sum(LetterValue[c] for c in word.upper())
